server/jobs/job_store.go

```diff
@@ -0,0 +1,144 @@
+package jobs
+
+import (
+	"fmt"
+	"sync"
+
+	"github.com/pkg/errors"
+)
+
+type JobStatus int
+
+const (
+	Processing JobStatus = iota
+	Complete
+)
+
+type Job struct {
+	Output []string
+	Status JobStatus
+}
+
+//go:generate pegomock generate -m --use-experimental-model-gen --package mocks -o mocks/mock_job_store.go JobStore
+
+type JobStore interface {
+	// Gets the job from the in memory buffer, if available and if not, reaches to the storage backend
+	// Returns an empty job with error if not in storage backend
+	Get(jobID string) (Job, error)
+
+	// Appends a given string to a job's output if the job is not complete yet
+	AppendOutput(jobID string, output string) error
+
+	// Sets a job status to complete and triggers any associated workflow,
+	// e.g: if the status is complete, the job is flushed to the associated storage backend
+	SetCompleteJobStatus(jobID string, status JobStatus) error
+
+	// Removes a job from the store
+	RemoveJob(jobID string)
+}
+
+func NewJobStore(storageBackend StorageBackend) JobStore {
+	return &LayeredJobStore{
+		jobs:           map[string]*Job{},
+		storageBackend: storageBackend,
+	}
+}
+
+// Setup job store for testing
+func NewTestJobStore(storageBackend StorageBackend, jobs map[string]*Job) JobStore {
```

Comment:
Can we set this up in the test file itself?

Reply:
Since the fields of the struct are private, we cannot access these fields outside of job_store.go! So, I think we need to set this up here unless we make the fields public!